### PR TITLE
refactor: incorporate `hyperswitch_interface` into router

### PR DESCRIPTION
**Pull request by Chethan-rao**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

This PR incorporates the `hyperswitch_interface` crate into the router crate. Secondly, all secrets(configs) decryption now happens during application startup, eliminating runtime decryption. Lastly, the addition of an encryption_client in the AppState enables easy runtime encryption and decryption processes.

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?

This PR refactors the way decryption of secrets work in router. So all flows should work fine wherever secrets are being used. 

Flows that has secrets decryption involved and requires testing  - 

* `merchant account` creation
* `api key` creation
* Hit `/health/ready` - Everything should be fine
* `Add card` `Retrieve card` `Delete card`
* `Currency conversion` flows
* `Apple pay flow` -  payment request with the decrypted apple pay token
*  Verify flow for applepay merchant verification 
* Payment method auth flow (Plaid)
* connector onboarding flow

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Review by SanchithHegde — COMMENTED** [review]
Can we also remove the `aws_kms`, `hashicorp-vault` and `aws_s3` features from the `router` crate, and directly pull the corresponding features from the `external_services` crate when the `release` feature is enabled in the `router` crate?

A similar change can be done in the `drainer` crate as well.

File: crates/router/src/configs/secrets_transformers.rs [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+#[async_trait::async_trait]
+impl SecretsHandler for settings::Database {
+    async fn convert_to_raw_secret(
+        value: SecretStateContainer<Self, SecuredSecret>,
+        secret_management_client: &dyn SecretManagementInterface,
+    ) -> CustomResult<SecretStateContainer<Self, RawSecret>, SecretsManagementError> {
+        let db = value.get_inner();
+        let db_password = secret_management_client
+            .get_secret(db.password.clone())
+            .await?;
```
[the remark this answers is not in the document]
Comment (Chethan-rao) [reply crates/router/src/configs/secrets_transformers.rs:19]: value gets moved if I use `into_inner()` and I can't call `transition_state`

File: crates/router/src/configs/settings.rs [hunk 2]
```diff
@@ -59,48 +55,34 @@ pub enum Subcommand {
...
-/// note: we can consider adding such behaviour in the future with custom implementation
-#[derive(Clone)]
-pub struct ActiveKmsSecrets {
-    pub jwekey: masking::Secret<Jwekey>,
-}
-
 #[derive(Debug, Deserialize, Clone, Default)]
 #[serde(default)]
-pub struct Settings {
+pub struct Settings<S: SecretState> {
```
[the remark this answers is not in the document]
Comment (Chethan-rao) [reply crates/router/src/configs/settings.rs:60]: making this private to crate will require lot of changes within the crate (visibility issues in lot of places) as `Settings` is being used in lot of places

File: crates/router/src/core/verification.rs [hunk 3]
```diff
@@ -13,42 +12,25 @@ pub async fn verify_merchant_creds_for_applepay(
     state: AppState,
     _req: &actix_web::HttpRequest,
```
[the remark this answers is not in the document]
Comment (prajjwalkumar17) [reply crates/router/src/core/verification.rs:13]: Yeah sure we can remove the req param.

**Review by SanchithHegde — APPROVED** [review]

**Review by SanchithHegde — DISMISSED** [review]

**Review by SanchithHegde — DISMISSED** [review]

**Review by SanchithHegde — DISMISSED** [review]

**Review by jarnura — DISMISSED** [review]

**Review by SanchithHegde — APPROVED** [review]

**Review by jarnura — APPROVED** [review]

**Pull request merged**